graphql_compiler/__init__.py

```diff
@@ -218,3 +224,57 @@ def graphql_to_redisgraph_cypher(schema, graphql_query, parameters, type_equival
+                object names. Columns will be represented as GraphQL fields and columns with
+                unsupported types will be ignored. Tables can be also represented as edges through
+                the junction_tables argument.
+        sql_edge_descriptors: dict, str-> SQLEdgeDescriptor, mapping identifiers for edges in
+                              the schema to namedtuples specifying the source and destination tables
+                              and which columns to use when traversing the edges. These edges will
+                              be rendered as vertex fields named out_<edgeIdentifier> and
+                              in_<edgeIdentifier> in the source and destination GraphQL objects
+                              respectively. The identifiers must not conflict with table
+                              identifiers.
+        junction_tables: dict, str -> JunctionTableEdgeDescriptor, mapping identifiers for junction
+                         table edges to namedtuples specifying how junction tables will
+                         be represented as edges in the schema. The identifiers must not conflict
+                         with table or edge identifiers.
+        dialect: sqlalchemy.engine.interfaces.Dialect, specifying the dialect we are compiling to
+                 (e.g. sqlalchemy.dialects.mssql.dialect()).
+        class_to_field_type_overrides: optional dict, class name -> {field name -> field type},
+                                       (string -> {string -> GraphQLType}). Used to override the
+                                       type of a field in the class where it's first defined and all
+                                       the class's subclasses.
+    Return:
+        SQLAlchemySchemaInfo containing the full information needed to compile SQL queries.
+    """
+    if not class_to_field_type_overrides:
+        class_to_field_type_overrides = {}
+
+    schema_graph = get_sqlalchemy_schema_graph(
+        tables, sql_edge_descriptors, junction_tables)
+
+    # I don't see a reason why someone would to hide a class using this parameter instead of not
+    # mentioning the class in the first place. This parameter is used to mostly deal with the
+    # complexities of omitting a class when classes inherit from it. Therefore, I do not expose
+    # it in the API.
+    hidden_classes = set()
+    graphql_schema = get_graphql_schema_from_schema_graph(
+        schema_graph, class_to_field_type_overrides, hidden_classes)
+
+    join_descriptors = get_restructured_edge_descriptors(sql_edge_descriptors)
+    # Note: We do not have inheritance in SQL so we should have type_equivalence_hints.
+    #       I think it's ok to have type_equivalence_hints to make the tests easier but let's not
+    #       expose those here.
+    type_equivalence_hints = {}
+    return make_sqlalchemy_schema_info(
+        graphql_schema, tables, join_descriptors, type_equivalence_hints, dialect)
```

Comment:
The `tables` in the schema info map GraphQL vertices to tables. Here you're giving a mapping from identifiers to tables, which are not the same.

Reply:
Since the identifiers will be used for vertex names based on the docstrings, then it will still be a "dict mapping every graphql object type or interface type name in the schema to a sqlalchemy table"